.gitignore

```diff
@@ -17,3 +17,4 @@
 
 .idea
 .pytest_cache
+.mypy_cache/
```

Comment:
What's this mypy? Could we benefit from using it?

Reply:
Mypy is basically the new builtin python type hinting, and vscode makes that cache folder for it for some reason. Mypy is pretty cool, especially for larger projects like ranger. You just add type hints primarily to function declarations and it helps your IDE debug typing issues. It doesn't enforce anything at run time, just helps your IDE. And you can add it incrementally. Although idk how well it would play with python2 or older versions of python3.

Reply:
So like @toonn said in the other comment: nice but I'd rather see it as a separate change not tied to this PR. It makes reverting stuff easier if such need arises.